src/Router.tsx

```diff
@@ -0,0 +1,37 @@
+import * as React from 'react';
+import {createBrowserRouter, useRouteError} from 'react-router-dom';
+import TeamOverview from './pages/TeamOverview';
+import Teams from './pages/Teams';
+import UserOverview from './pages/UserOverview';
+
+function ErrorBoundary() {
+    const error: any = useRouteError();
+    return (
+        <div>
+            <p>Something went wrong.</p>
+            {error.message && <span>Here is the error: {error.message}</span>}
+        </div>
+    );
+}
+
+export default createBrowserRouter([
+    {
+        path: '/',
+        element: <Teams />,
+        errorElement: <ErrorBoundary />,
```

Comment:
Added error boundary and unexpected error handling. we can make this component more interactive as well.